packages/components/pie-button/src/index.ts

```diff
@@ -43,6 +58,95 @@ export class PieButton extends LitElement implements ButtonProps {
+
+    @property({ attribute: 'formaction' })
+    public formAction: ButtonProps['formAction'];
+
+    @property({ attribute: 'formenctype' })
+    public formEncType: ButtonProps['formEnctype'];
+
+    @property({ attribute: 'formmethod' })
+    public formMethod: ButtonProps['formMethod'];
+
+    @property({ attribute: 'formnovalidate', type: Boolean })
+    public formNoValidate: ButtonProps['formNoValidate'];
+
+    @property({ attribute: 'formtarget' })
+    public formTarget: ButtonProps['formTarget'];
+
+    /**
+     * This method creates an invisible button of the same type as pie-button. It is then clicked, and immediately removed from the DOM.
+     * This is done so that we trigger native form actions, such as submit and reset in the browser. The performance impact of adding and removing a single button to the DOM
+     * should be neglible, however this should be monitored.
+     * This is the only viable way of guaranteeing native button behaviour when using a web component in place of an actual HTML button.
+     *
+     * TODO: if we need to repeat this logic elsewhere, then we should consider moving this code to a shared class or mixin.
+     */
+    private _simulateNativeButtonClick (btnType: 'submit' | 'reset') {
+        if (!this.form) return;
+
+        const btn = document.createElement('button');
+        btn.type = btnType;
+
+        // Visually hidden styles
+        btn.style.position = 'absolute';
+        btn.style.width = '1px';
+        btn.style.height = '1px';
+        btn.style.padding = '0';
+        btn.style.margin = '-1px';
+        btn.style.overflow = 'hidden';
+        btn.style.border = '0';
+        btn.style.whiteSpace = 'nowrap';
+
+        if (btnType === 'submit') {
+            if (this.name) {
+                btn.name = this.name;
+            }
+            if (this.value) {
+                btn.value = this.value;
+            }
+            if (this.formAction) {
+                btn.setAttribute('formaction', this.formAction);
+            }
+            if (this.formEncType) {
+                btn.setAttribute('formenctype', this.formEncType);
+            }
+            if (this.formMethod) {
+                btn.setAttribute('formmethod', this.formMethod);
+            }
+            if (this.formNoValidate) {
+                btn.setAttribute('formnovalidate', 'formnovalidate');
+            }
+            if (this.formTarget) {
+                btn.setAttribute('formtarget', this.formTarget);
+            }
+        }
+
+        this.form.append(btn);
+        btn.click();
+        btn.remove();
+    }
+
+    private _handleClick () {
+        if (!this.isLoading && this.form) {
+            if (this.type === 'submit') {
+                // only submit the form if either formnovalidate is set, or the form passes validation checks (triggers native form validation)
+                if (this.formNoValidate || (!this.formNoValidate && this.form.reportValidity())) {
```

Comment:
I think this is equivalent?
```suggestion
                if (this.formNoValidate || this.form.reportValidity()) {
```

Reply:
lovely spot - pushing now